bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/Watchdog.java

```diff
@@ -95,20 +95,24 @@ public Watchdog(Clock clock, long waitTimeoutMs, long idleTimeoutMs) {
   }
 
   public void run() {
-    Iterator<Entry<WatchedCall<?, ?>, Object>> it = openStreams.entrySet().iterator();
+    try {
+      Iterator<Entry<WatchedCall<?, ?>, Object>> it = openStreams.entrySet().iterator();
 
-    int count = 0;
+      int count = 0;
 
-    while (it.hasNext()) {
-      WatchedCall<?, ?> stream = it.next().getKey();
-      if (stream.cancelIfStale()) {
-        count++;
-        it.remove();
+      while (it.hasNext()) {
+        WatchedCall<?, ?> stream = it.next().getKey();
+        if (stream.cancelIfStale()) {
+          count++;
+          it.remove();
+        }
       }
-    }
 
-    if (count > 0) {
-      LOG.warn("Found %d stale streams and cancelled them", count);
+      if (count > 0) {
+        LOG.warn("Found %d stale streams and cancelled them", count);
+      }
+    } catch (Throwable t) {
+      LOG.error("Caught throwable in periodic Watchdog run. Continuing.", t);
```

Comment:
I think it would be cleaner to move error handling out of this method, something like:

run() {
  try {
   runUnsafe();
  } catch (...
}

runUnsafe() {
  while(..)
}

Reply:
Done